addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.js

```diff
@@ -161,14 +282,82 @@ class ComposerTextInput extends Component {
      */
     _onKeydownTextarea(ev) {
         switch (ev.key) {
+            // UP, DOWN, TAB: prevent moving cursor if navigation in mention propositions
+            case 'ArrowUp':
+            case 'ArrowDown':
+            case 'Tab':
+                if (this.state.isMentionsOpen) {
+                    ev.preventDefault();
+                }
+                break;
+            // ENTER: submit the message only if the dropdown mention proposition is not displayed
             case 'Enter':
-                this._onKeydownTextareaEnter(ev);
+                if (this.state.isMentionsOpen) {
+                    ev.preventDefault();
+                } else {
+                    this._onKeydownTextareaEnter(ev);
+                }
                 break;
+        }
+    }
+
+        /**
+     * @private
+     * @param ev
+     */
+    _onKeyupTextarea(ev) {
+        switch (ev.key) {
+            // ESCAPE: close mention propositions
             case 'Escape':
-                this._onKeydownTextareaEscape(ev);
+                if (this.state.isMentionsOpen) {
+                    this.state.isMentionsOpen = false;
+                    this.composer.unlink({ mentionedPartners: null });
+                    ev.stopPropagation();
+                } else {
+                    this._onKeyupTextareaEscape(ev);
```

Comment:
doesn't make sense that this function suggests by its naming it handles ESC, but in reality it just does this when mention is not open. Either include logic of mention open in this function, or rename function (I would choose 1st option).